packages/javascript-sdk/src/oauth2-client/index.ts

```diff
@@ -160,6 +160,7 @@ abstract class OAuth2Client {
     };
 
     const response = await this.request('accessToken', undefined, false, init, options);
+    const responseClone = response.clone();
```

Comment:
Cloning the response here since we want to both grab the response text as well as create parsed JSON out of it.

Reply:
Can you point me to what you mean? I don't understand why the `clone` is necessary?

Reply:
Good question! This is something that took me a bit to wrap my head around (I had to do this a lot in the Token Vault plugin). The tl;dr of is any time you call a method off of `body`, you "attach" a "reader" to the underlying `ReadableStream` of the `body`. Only one reader can be attached to a stream at a time, so if you have two different places in the code that need to "attach" to the stream of the `body`, then you have to `clone` it.

The other location where we read the `body` is here: https://github.com/ForgeRock/forgerock-javascript-sdk/blob/develop/packages/javascript-sdk/src/oauth2-client/index.ts#L301.

Ref for the above: https://developer.mozilla.org/en-US/docs/Web/API/Streams_API/Using_readable_streams#attaching_a_reader

Reply:
wow havent used this before at all. interesting stuff